autoscheduler/frontend/src/components/LastUpdatedAt.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from 'react';
+import { Typography } from '@material-ui/core';
+import { useSelector } from 'react-redux';
+import { RootState } from '../redux/reducer';
+
+/** Given a date, formats it into a string for how long ago it was.
+ *  Note that now is only included for mocking the current time for tests.
+ */
+export function getLastUpdatedAtText(prev: Date, now = new Date()): string {
+  const seconds = (now.getTime() - prev.getTime()) / 1000;
+
+  // If it was less than 60 minutes ago, say it was x minutes ago
+  const minutes = seconds / 60;
+  if (minutes < 60) {
+    return `${minutes.toFixed(0)} minutes ago`;
+  }
+
+  // If it was less than 24 hours ago, say it was x hours ago
+  const hours = minutes / 60;
+  if (hours < 24) {
+    return `${hours.toFixed(0)} hours ago`;
+  }
+
+  // If it was less than a week ago, say it was x days ago
+  const days = hours / 24;
+  if (days < 7) {
+    return `${days.toFixed(0)} days ago`;
+  }
+
+  // If it was more than a week ago, say it was on x date (e.g. 11/15/20)
+  const year = `${prev.getFullYear().toFixed(0)}`.substr(2);
+  const month = prev.getMonth() + 1; // months are 0-indexed
+  const day = prev.getDate(); // gets the day of the month
+
+  return `on ${month}/${day}/${year}`;
+}
+
+/**
+ * Displays when the current term was last updated, e.g. "Last updated 3 hours ago."
+ */
+const LastUpdatedAt: React.FC = () => {
+  const [lastUpdated, setLastUpdated] = React.useState<Date>();
+  const [lastUpdatedText, setLastUpdatedText] = React.useState('');
+  const term = useSelector<RootState, string>((state) => state.term);
+  // Store the last term so we can check if it changed
+  const [lastTerm, setLastTerm] = React.useState(term);
+
+  React.useEffect(() => {
+    const fetchLastUpdated = (): void => {
+      if (!term) {
+        // So the last updated text disappears on the landing page
+        setLastUpdated(null);
+        setLastUpdatedText('');
+        return;
+      }
+
+      // If the last term changed, then set lastUpdated to null so it disappears while the new one
+      // is loading
+      if (term !== lastTerm) {
+        setLastUpdated(null);
+        setLastUpdatedText('');
+      }
+
+      setLastTerm(term);
+
+      fetch(`api/get_last_updated?term=${term}`).then((res) => res.json()).catch(() => '').then((dateStr) => {
+        if (!dateStr) {
+          setLastUpdated(undefined);
+          setLastUpdatedText('');
+          return;
+        }
+
+        const date = new Date(dateStr);
+        setLastUpdated(date);
+        setLastUpdatedText(getLastUpdatedAtText(date));
+      });
+    };
+
+    fetchLastUpdated();
+
+    const interval = setInterval(() => fetchLastUpdated, 5 * 60 * 1000); // update every 5 minutes
```

Comment:
This doesn't actually periodically call the fetch again, it seems to do with the fact that you call `fetchLastUpdated` instead of doing the logic in the arrow function since it works when I remove the function definition. I'm not sure exactly why your current implementation doesn't work though, as far as I know it should be fine but there are probably some javascript shenanigans happening behind the scenes

Reply:
Changed it from `fetchLastUpdated` to `fetchLastUpdated()` fixed it, although I don't really know why exactly. JS is weird

Reply:
Oh I didn't notice that, without the parens at the end the arrow function just returns `fetchLastUpdated` instead of calling it. You could have done `setInterval(fetchLastUpdated, 5 * 60 * 1000)` to get the same effect